### PR TITLE
GC: Restore: Disable Item.Attachment Restore

### PR DESCRIPTION
**Pull request by dadams39**
## Description

`Item.Attachments` have unique properties that require transformation to be uploaded into the M365 system. All objects of this type are temporarily prevented from uploading via this patch. As Issue #2353 is addressed, there will be more coverage for the various types of item attachments. 

## Does this PR need a docs update or release note?

- [x] :white_check_mark: Yes, it's included

## Type of change

- [x] :sunflower: Feature

## Issue(s)

*related to  #2353<issue>

## Test Plan

- [x] :muscle: Manual

### CODE REVIEW

File: src/internal/connector/exchange/attachment.go [hunk 1]
```diff
@@ -53,6 +53,12 @@ func uploadAttachment(
 		return nil
 	}
 
+	// item Attachments to be skipped until the completion of Issue #2353
+	if attachmentType == models.ITEM_ATTACHMENTTYPE {
+		logger.Ctx(ctx).Debugf("skip uploading of item attachment. Not supported: ", *attachment.GetName())
```
Comment (vkamra) [inline src/internal/connector/exchange/attachment.go:58]: @dadams39 - let's make this an Info log. @ryanfkeepers - what "clues" would be useful to add here? You'd want the user to be able to identify exactly which email they hit this issue with

Comment (ryanfkeepers) [reply src/internal/connector/exchange/attachment.go:58]: `logger.Ctx(ctx)` will import all clues currently in the context automatically.  That means if the caller already set the item ID into the context, it gets inherited here.  For ad-hoc values, I'd suggest the following:
```suggestion
		logger.Ctx(ctx).Debugw("item attachment uploads are not supported ",
		    "attachment_name", *attachment.GetName(),
		    "attachment_type", attachmentType,
		    "attachment_id", *attachment.GetId())
```

Comment (dadams39) [reply src/internal/connector/exchange/attachment.go]: Resolved with [962aaf8](https://github.com/alcionai/corso/pull/2359/commits/962aaf895ac48e8130d6ad2cac4b3f787153c56a)

**Review by ryanfkeepers — APPROVED** [review]

**Comment by vkamra** [comment]
@dadams39 - let's please add a mock test here before merging. I want to validate that restore will indeed not fail.

**Comment by dadams39** [comment]
@vkamra Test added within [1d5c1ac](https://github.com/alcionai/corso/pull/2359/commits/1d5c1acc6ff11ed60ca67f86036c89cf5a97c03a)

NOTE test would fail on branch #2353 without this patch
```bash
go test -v . --run TestExchangeRestoreSuite/TestRestoreExchangeObject/Test_Mail:_Item_Attachment
=== RUN   TestExchangeRestoreSuite
=== RUN   TestExchangeRestoreSuite/TestRestoreExchangeObject
=== RUN   TestExchangeRestoreSuite/TestRestoreExchangeObject/Test_Mail:_Item_Attachment
--- PASS: TestExchangeRestoreSuite (2.10s)
    --- PASS: TestExchangeRestoreSuite/TestRestoreExchangeObject (2.10s)
        --- PASS: TestExchangeRestoreSuite/TestRestoreExchangeObject/Test_Mail:_Item_Attachment (2.09s)

```

**Comment by vkamra** [comment]
Excellent. Added mergequeue to get this in

**Pull request merged**